apps/inference/tests/conftest.py

```diff
@@ -1,2 +1,68 @@
-# conftest.py
-# This file can be empty and still affect pytest's import resolution
+import asyncio
+import gc
+import json
+import os
+
+import pytest
+import torch
+from fastapi.testclient import TestClient
+
+import neuronpedia_inference.server as server
+from neuronpedia_inference.args import parse_env_and_args
+from neuronpedia_inference.config import Config
+from neuronpedia_inference.sae_manager import SAEManager
+from neuronpedia_inference.server import app, initialize
+from neuronpedia_inference.shared import Model
+
+BOS_TOKEN_STR = "<|endoftext|>"
+TEST_PROMPT = "Hello, world!"
+
+
+@pytest.fixture(scope="session")
+def initialize_models():
+    """
+    Defining the global state of the app with a session-scoped fixture that initializes the model and SAEs.
+
+    This fixture will be run once per test session and will be available to all tests
+    that need an initialized model. It uses the same initialization logic as the
+    /initialize endpoint.
+    """
+    # Set environment variables for testing
+    os.environ.update(
+        {
+            "MODEL_ID": "gpt2-small",
+            "SAE_SETS": json.dumps(["res-jb"]),
+            "MODEL_DTYPE": "float16",
+            "SAE_DTYPE": "float32",
+            "TOKEN_LIMIT": "500",
+            "DEVICE": "cpu",
+            "INCLUDE_SAE": json.dumps(
+                ["7-res-jb"]
+            ),  # Only load the specific SAE we want
+            "EXCLUDE_SAE": json.dumps([]),
+            "MAX_LOADED_SAES": "1",
+            "SECRET": "cat",  # Match the secret key used in tests
```

Comment:
Nitpick: Possibly make this a shared variable like TEST_PROMPT so it can be shared with test_activation_all